src/sky.cpp

```diff
@@ -204,338 +202,454 @@ void Sky::render()
 
 		video::SColor cloudyfogcolor = m_bgcolor;
 
-		// Draw far cloudy fog thing blended with skycolor
-		for (u32 j = 0; j < 4; j++) {
-			video::SColor c = cloudyfogcolor.getInterpolated(m_skycolor, 0.45);
-			vertices[0] = video::S3DVertex(-1, 0.08, -1, 0, 0, 1, c, t, t);
-			vertices[1] = video::S3DVertex( 1, 0.08, -1, 0, 0, 1, c, o, t);
-			vertices[2] = video::S3DVertex( 1, 0.12, -1, 0, 0, 1, c, o, o);
-			vertices[3] = video::S3DVertex(-1, 0.12, -1, 0, 0, 1, c, t, o);
-			for (video::S3DVertex &vertex : vertices) {
-				if (j == 0)
-					// Don't switch
-					{}
-				else if (j == 1)
-					// Switch from -Z (south) to +X (east)
-					vertex.Pos.rotateXZBy(90);
-				else if (j == 2)
-					// Switch from -Z (south) to -X (west)
-					vertex.Pos.rotateXZBy(-90);
-				else
-					// Switch from -Z (south) to +Z (north)
-					vertex.Pos.rotateXZBy(-180);
+		if (m_dynamic_visible) {
+			// Draw far cloudy fog thing blended with skycolor
+			for (u32 j = 0; j < 4; j++) {
+				video::SColor c = cloudyfogcolor.getInterpolated(m_skycolor, 0.45);
+				vertices[0] = video::S3DVertex(-1, 0.08, -1, 0, 0, 1, c, t, t);
+				vertices[1] = video::S3DVertex( 1, 0.08, -1, 0, 0, 1, c, o, t);
+				vertices[2] = video::S3DVertex( 1, 0.12, -1, 0, 0, 1, c, o, o);
+				vertices[3] = video::S3DVertex(-1, 0.12, -1, 0, 0, 1, c, t, o);
+				for (video::S3DVertex &vertex : vertices) {
+					if (j == 0)
+						// Don't switch
+						{}
+					else if (j == 1)
+						// Switch from -Z (south) to +X (east)
+						vertex.Pos.rotateXZBy(90);
```

Comment:
A lookup table is more appropriate for this task.

Reply:
Ah, no. We can't use a lookup table because we're only drawing the sides on the regular skybox.

Reply:
Actually I suggested to use a `j` to `degrees` conversion table, but keep it like this since the comments are helpful.